### PR TITLE
Add detection of mail addresses

### PR DESCRIPTION
**Pull request by heiCode**
* Automatically convert mail addresses to links.
* Add options to set custom regular expressions for mail address detection
* Update docs

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -23,62 +25,81 @@ export default class MagicUrl {
...
+          newDelta.insert(str)
+        } else if (mailMatches && mailMatches.length) {
+          mailMatches.forEach(match => {
+            const split = str.split(match)
+            const beforeLink = split.shift()
+            newDelta.insert(beforeLink)
+            newDelta.insert(match, {link: `mailto:${match}`})
+            str = split.join(match)
+          })
+          newDelta.insert(str)
```
Comment (visualjerk) [inline src/index.js:49]: This section could possible be more easy to read, if we abstract the match handling.

Something along those lines:

```js
function addMatchToDelta(match, normalizer) {
  const split = str.split(match)
  const beforeLink = split.shift()
  newDelta.insert(beforeLink)
  newDelta.insert(match, {link: normalizer(match)})
  str = split.join(match)
}

if (urlMatches && urlMatches.length) {
  urlMatches.forEach(match => addMatchToDelta(match, this.urlNormalizer))
} else if (mailMatches && mailMatches.length) {
  mailMatches.forEach(match => addMatchToDelta(match, this.mailNormalizer))
}
```

**Review by visualjerk — CHANGES_REQUESTED** [review]
Pretty nice 👍  Thanks for this great addition.

File: src/index.js (hunk 1 above)
Comment (heiCode) [reply src/index.js]: Hi @visualjerk, thanks for your review! I've implmented the requested changes and updated the branch. Looking forward to get your feedback!

Comment (visualjerk) [reply src/index.js]: Perfect 😃 Thanks again!

**Comment by visualjerk** [comment]
This is released in version 3.0.0

**Pull request merged**